kore/src/SMT.hs

```diff
@@ -622,6 +481,126 @@ setOption :: MonadSMT m => Text -> SExpr -> m ()
 setOption infoFlag expr =
     ackCommand $ List (Atom "set-option" : Atom infoFlag : [expr])
 
+{- | Query an external SMT solver (if available).
+
+The solver may be shared among multiple threads. Individual commands will
+acquire and release the solver as needed, but sequences of commands from
+different threads may be interleaved; use 'withSolver' to acquire exclusive
+access to the solver for a sequence of commands.
+-}
+newtype SMT a = SMT (Maybe SolverSetup -> LoggerT IO a)
```

Comment:
Just wondering, what's the reasoning behind making this explicit instead of using `ReaderT`?

Reply:
No particular reason, just a stylistic preference.